### PR TITLE
replace add-path with GITHUB_ENV

### LINKED ISSUE #3446: add-path is deprecated - use env files for CI
`add-path` is deprecated - should be using env files for CI
https://github.blog/changelog/2020-10-01-github-actions-deprecating-set-env-and-add-path-commands/

### PR DESCRIPTION
**Pull request by xxyzz**
close #3446

### CODE REVIEW

**Comment by pmatos** [comment]
@xxyzz thanks for the fix. `ci-ubsan` has another occurrence of `add-path`, can you please fix?
Also the reason I didn't use `GITHUB_WORKSPACE` it's because it was broken in github runner. It might have been fixed already. If possible, please fix the other `add_path`. This will trigger a CI run and allows us to check if github has fixed the `GITHUB_WORKSPACE` references inside the docker container.

File: .github/workflows/ci-ubsan.yml [hunk 1]
```diff
@@ -21,12 +21,8 @@ jobs:
         export cpus=$(grep -c ^processor /proc/cpuinfo)
         make CPUS="${cpus}" PKGS="racket-test db-test unstable-flonum-lib net-test" CONFIGURE_ARGS_qq='CFLAGS="-fno-var-tracking-assignments" --enable-ubsan' RACKETBC_SUFFIX="" bc-in-place 2>&1 | tee logs/build.log
     - name: Add new racket to PATH
-      # This should be:
-      # run: echo "::add-path::${{ github.workspace }}/racket/bin"
-      # but due to a github bug, the variable ${{ github.workspace }} value inside a container is broken
-      # so we hard code the real value
       run: |
-        echo "::add-path::/__w/racket/racket/racket/bin"
```
Comment (pmatos) [inline .github/workflows/ci-ubsan.yml:29]: Check line 115 as well pls.

**Review by pmatos — CHANGES_REQUESTED** [review]

**Comment by xxyzz** [comment]
It works fine on my forked repo so I guess the bug was fixed.

**Comment by pmatos** [comment]
> It works fine on my forked repo so I guess the bug was fixed.

It seems the bug is fixed indeed. For the record this was the problem: https://github.community/t/runner-temp-being-emptied-between-jobs/18217/2?u=pmatos

I will merge. Thanks.

**Pull request merged**